app/controllers/api/v3/courses_controller.rb

```diff
@@ -24,11 +36,19 @@ def show
 
     private
 
+      # Used for sitemap on Find
+      # When these fields are specified return all results for sitemap
+      def disable_pagination?
+        if params[:fields] && params[:fields][:courses]
+          (params[:fields][:courses].split(",") & %w[course_code provider_code updated_at]).size == 3
+        end
+      end
```

Comment:
Disabling pagination via a specific combination of `fields` feels... odd to me? I'd expect something more explict and less surprising but i'm not sure what it could be. 

At the very least I think this could read better if the check was something along the lines of:

```ruby
params[:fields][:courses].split(",") == fields_required_To_disable_pagination
``` 

Or something similarly explict

Reply:
I think I agree. I had in mind an approach where the user could disable pagination / increase the page size over a threshold, and if so requested it would then ensure that the fields requested are appropriate (any of `%w[course_code provider_code updated_at]`). If any other fields are requested, it would raise a 4XX error, perhaps with info about which fields can/can't be requested.

Reply:
I've have made @danielburnley suggestion of extracting to a method. Although `==` cannot be used as ordering is not guaranteed.

As for the more general note of how we disable/enable pagination and flow I think this needs to be done at a higher level overall otherwise we'll get shotgun snippets of code like this everywhere. That I think is outside the scope of this ticket

Reply:
Why is this outside the scope of this ticket? This ticket is about that very thing, being able to disable pagination, so we should implement it in a fitting way, unless it's a big enough piece of work that warrants being pulled out.

Reply:
If the client wants to disable pagination they should be able to add a flag/param for this which we should respect. If the client wants fewer or custom fields they should ask for that too.

If we feel like we're giving out too much info (fields-wise) I think that means that the data should be on another endpoint which the client would have to make a request to.

If we're concerned about who does this then ie which clients. may this should be done at an authorisation level eg. only these sets of api keys can disable pagination as a feature

If we're concerned about the amount of data sent because it's the whole dataset and therefore performance issues then I think pagination should not be disabled and the client will have to simply paginate through the dataset

Reply:
I'm not sure I understand why this means that we should implement this as a magical "we'll disable pagination if you specify the right fields" approach.

My point is that the client should be in charge of defining the request it's making, and it shouldn't get _ALL_ the results back if it happens to decide that it only needs `course_code`, `provider_code` and `updated_at`. That would probably be surprising, and being a public API we shouldn't make assumptions about how it's used, and try to be as unsurprising as possible.

Reply:
Ok, I see. I can do `?paginate=false` defaults to true and only if these fields are specified then disable pagination. My point is that the feature of disabling pagination should that only be applicable for this endpoint? Otherwise we'll have other areas where we'll end up having to do the same thing

Reply:
Ya, so that's a good point that this mechanism should be more widely defined. I'd be happy if we hacked this in for now with `per_page=65535` or something like that. At least it'll be less surprising.